Automattic/wp-calypso
# Checkout: Disable required fields

#### Changes proposed in this Pull Request

This is an experiment to see what happens if no domain contact fields in checkout are marked (on the client) as required. In theory, all contact validation should occur on the server during the validation call, so these required fields should be unnecessary (and may in fact cause problems like the one described in https://github.com/Automattic/wp-calypso/pull/55882).

If this is successful, we can also remove all the `isRequired` machinery, validation, and data fields.

#### Testing instructions

The basic goal here is to see if there's any way to submit invalid contact form data and _not_ get an error.

1. Visit checkout with a `.ca` domain in the cart.
1. Fill out the contact form, selecting "Canadian Citizen" as the entity type.
1. Do not check the "I have read" checkbox at the bottom of the contact form.
1. Click "Continue" on the form.
1. Verify that you receive a validation error.
1. Check the checkbox and resubmit the form.
1. Verify that the step completes successfully.
1. Edit the contact form again and select a country that requires a postal code (most of them).
1. Leave the postal code field blank and try to submit the form.
2. Verify that you get a validation error.
3. Enter a nonsensical postal code and try to submit the form.
2. Verify that you get a validation error.
3. Select a country that does not have a postal code (eg: Curacao).
4. Submit the form and verify that there are no errors.
